packages/types/src/types.ts

```diff
@@ -124,6 +124,7 @@ export interface IMethod extends Codec {
 
 export interface IExtrinsicSignature extends Codec {
   readonly isSigned: boolean;
+  era: Codec;
```

Comment:
Not quite correct here - this means it can be any Codec, so have the interface defined (which would then be implemented by the Era classes)

Reply:
Thanks.. Can you please suggest some interface. We used Codec cause we didn't find anything in common in Mortal and Immortal Era which we could place in interface and moreover its only used in ExtrinsicSignature to encode U8a

Reply:
Well, `era` is an `ExtrinsicEra` - possibly `IExtrinsicEra` (with `asMortal` & `asImmortal`) - that is if it is actually used where the interface is passed.

Reply:
Sure we will make this change as you said, actually we've thought about that, only problem is the return type of asMortal and asImmortal would be a real implementation class, which seems against the reason we use interface here.

Reply:
Yes, that would be the quick one - which doesn't require adding yet more.